storage/remote/queue_manager.go

```diff
@@ -398,6 +443,37 @@ func (t *QueueManager) Stop() {
 	t.metrics.unregister()
 }
 
+// Getting the data from a shard and structuring it as a SegmentRecord from a shard
+func (t *QueueManager) getSegmentRecords() SegmentRecord {
+	RecordSegment := t.shards.qm.watcher.SegmentFile
+	RecordEndpoint := t.shards.qm.url
+
+	record := SegmentRecord{
+		Segment:  RecordSegment,
+		Endpoint: RecordEndpoint,
+	}
+
+	return record
+}
+
+// Writes the segment record in a JSON file
+func (t *QueueManager) RecordSegment() {
```

Comment:
Does `RecordSegment()` need to be exposed?

Reply:
What do you mean by exposed?

Reply:
It's like public vs priavte from other languages. If the name of a function starts with a lower case, `recordSegment`, it's only available within the package it's defined. In this case, `storage/remote`. If the name of a function starts with an upper case, `RecordSegment`, it's available outside that package to anyone who imports that package.